credentials/src/main/kotlin/web5/sdk/credentials/model/VcDataModel.kt

```diff
@@ -0,0 +1,346 @@
+  override fun serialize(value: Date?, gen: JsonGenerator?, serializers: SerializerProvider?) {
+    gen?.writeString(value?.let { DATE_FORMAT.format(it) })
+  }
+}
+
+private class DateDeserializer : JsonDeserializer<Date>() {
+  override fun deserialize(p: JsonParser?, ctxt: DeserializationContext?): Date {
+    return DATE_FORMAT.parse(p?.text ?: "")
+  }
+}
+
+private class CredentialSubjectSerializer : JsonSerializer<CredentialSubject>() {
+  override fun serialize(value: CredentialSubject, gen: JsonGenerator, serializers: SerializerProvider) {
+    gen.writeStartObject()
+    // Write the id field. If id is null, write an empty string; otherwise, write its string representation.
+    gen.writeStringField("id", value.id?.toString() ?: "")
+    value.additionalClaims.forEach { (key, claimValue) ->
+      gen.writeObjectField(key, claimValue)
+    }
+    gen.writeEndObject()
+  }
+}
+
+private class CredentialSubjectDeserializer : JsonDeserializer<CredentialSubject>() {
+  override fun deserialize(p: JsonParser, ctxt: DeserializationContext): CredentialSubject {
+    val node: JsonNode = p.codec.readTree(p)
+    val idNode = node.get("id")
+    val id = idNode?.asText()?.takeIf { it.isNotBlank() }?.let { URI.create(it) }
+
+    // Remove the "id" field and treat the rest as additionalClaims
+    val additionalClaims = node.fields().asSequence().filterNot { it.key == "id" }
+      .associate { it.key to it.value.asText() } // Assuming all values are stored as text
+
+    return CredentialSubject.Builder()
+      .id(id)
+      .additionalClaims(additionalClaims)
+      .build()
+  }
+}
+
+private fun getObjectMapper(): ObjectMapper {
+  return jacksonObjectMapper().apply {
+    registerKotlinModule()
+    setSerializationInclusion(JsonInclude.Include.NON_NULL)
+
+    val dateModule = SimpleModule().apply {
+      addSerializer(Date::class.java, DateSerializer())
+      addDeserializer(Date::class.java, DateDeserializer())
+      addSerializer(CredentialSubject::class.java, CredentialSubjectSerializer())
+      addDeserializer(CredentialSubject::class.java, CredentialSubjectDeserializer())
+    }
+    registerModule(dateModule)
+  }
+}
+
+/**
+ * The [VcDataModel] instance representing the core data model of a verifiable credential.
+ *
+ * @see {@link https://www.w3.org/TR/vc-data-model/#credentials | VC Data Model}
+ */
+public class VcDataModel(
+  public val id: URI?,
+  @JsonProperty("@context")
+  public val context: List<URI>,
+  public val type: List<String>,
+  public val issuer: URI,
+  public val issuanceDate: Date,
+  public val expirationDate: Date?,
+  public val credentialSubject: CredentialSubject,
+  public val credentialSchema: CredentialSchema?,
+  public val credentialStatus: StatusList2021Entry?
+) {
+  /**
+   * Builder class for creating [VcDataModel] instances.
+   */
+  public class Builder {
+    private var id: URI? = null
+    private var context: MutableList<URI> = mutableListOf()
```

Comment:
context and type don't have to be mutable, right? they just get set, never modified?

Reply:
The builder can add things to them if it is missing the default:
```
      // Default context and type handling
      if (context.isEmpty() || context[0].toString() != DEFAULT_VC_CONTEXT) {
        context.add(0, URI.create(DEFAULT_VC_CONTEXT))
      }

      if (type.isEmpty() || type[0] != DEFAULT_VC_TYPE) {
        type.add(0, DEFAULT_VC_TYPE)
      }
```

Reply:
I updated this to match the spec gabe made, will throw if Defaults are not in now instead of dynamically adding them

Reply:
awesome, then this field type can just be a `List<URI>` now?

Reply:
derp, you changed it already in your last commit!